### ISSUE TITLE
put into spec team, but can kill, and spawn in enemy's team

### ISSUE THREAD

**Issue by ScalaWilliam** [ActionFPS/ActionFPS-Game#144]
![20170206_11 55 32_ac_complex_tdm](https://cloud.githubusercontent.com/assets/2464813/22647319/c47d84fe-ecab-11e6-9f70-5a80b6bb0375.jpg)

one of these two demos should contain
https://cdn.discordapp.com/attachments/231380080591372288/278141903600484352/20170206_1203_62.210.131.155_ac_depot_4min_CTF.dmo
https://cdn.discordapp.com/attachments/231380080591372288/278141909078245377/20170206_1216_62.210.131.155_ac_depot_10min_CTF.dmo

**Comment by gyunderscorebe** [comment]
What's server/client commit?

**Comment by lucasgautheron** [comment]
@gyunderscorebe master.
This is probably after my update to set recover teams of reconnecting clients. I was curious to see how it would work. Turns out the game doesn't send initial team to the reconnecting client but only to others. So it's all about letting the reconnecting client know which team he is on.

**Comment by ScalaWilliam** [comment]
I was on https://github.com/ActionFPS/ActionFPS-Game/pull/125 when it was happening though

**Comment by lucasgautheron** [comment]
Oh so master wasn't merged into it;

**Comment by lucasgautheron** [comment]
What matters is that the server was compiled from master

**Comment by ScalaWilliam** [comment]
ah server side prob

**Comment by tejlibre** [comment]
@lucasgautheron do we know what commits introduced the bug? If we have no idea when the bug was introduced, we could test a bunch of releases say, between december and february and get a starting point that way

**Comment by lucasgautheron** [comment]
@tejlibre I'd say this one : https://github.com/ActionFPS/ActionFPS-Game/commit/fcf28cfe74937c2450347981f868d972a50fd9fd

**Comment by ScalaWilliam** [comment]
I know it's fairly vague for somebody not familiar with this game, but the help would be very appreciated.

You can get test users [from our Discord](https://discord.gg/YSAgM76). You'll probably want to have some C/C++ experience for doing this.

I can't support helping fix this problem, but can help in testing the final changes. Sorry, not enough time for me.

**Comment by ScalaWilliam** [comment]
I posted this on BountySource: https://www.bountysource.com/issues/41713635-put-into-spec-team-but-can-kill-and-spawn-in-enemy-s-team

**Comment by spelcaster** [comment]
Hi there, so I've started to study this issue and I think I'm very close to find the culprit.

First, I thought that this could be happening on the server side, I was suspecting that it was doing a weak comparison between clients in the same host, but the clients have some kind of unique id that is called `clientnum`. The sender argument, [here](https://github.com/ActionFPS/ActionFPS-Game/blob/master/source/src/server.cpp#L2709) is the same as the `clientnum` [here](https://github.com/ActionFPS/ActionFPS-Game/blob/master/source/src/server.h#L109) and [here](https://github.com/ActionFPS/ActionFPS-Game/blob/master/source/src/entity.h#L425). So this problem was less likely to be on server side.

My second assumption was that it could be something wrong with the functions that render the scoreboards, I did some tweaks and discovered that the second client does not update the playerent, but the first client is doing everything fine (apparently).

So, basically, there's a condition somewhere that is preventing the client to update the playerent.

[scoreboard-player-dump.txt](https://github.com/ActionFPS/ActionFPS-Game/files/1807833/scoreboard-player-dump.txt)

**Comment by spelcaster** [comment]
@ScalaWilliam, I've found the bug and fixed it, I'll send the commit later.
